concert/ext/ufo.py

```diff
@@ -282,6 +296,150 @@ def z_parameter(self, name):
         self._z_parameter = name
 
 
+class QuantifiedProxyArgs(Parameterizable):
+
+    """Data class holding arguments for tofu with units."""
+
+    UNITS = {
+        'source_position_y': q.px,
+        'detector_position_y': q.px,
+        'center_position_x': q.px,
+        'center_position_z': q.px,
+        'axis_angle_x': q.rad,
+        'energy': q.keV,
+        'propagation_distance': q.m,
+        'pixel_size': q.m,
+        'retrieval_padded_width': q.px,
+        'retrieval_padded_height': q.px,
+        'projection_margin': q.px,
+        'x_region': q.px,
```

Comment:
Off-topic: Since years I'm thinking about a elegant way to use vectors/lists in Quantities/Properties. At some point somebody tries to do `args.x_region[2] = 10*q.px` ...

Reply:
Doesn't that work like it should automatically?

Reply:
No, the getter will give you the vector (or whatever), but the setter will not be called.

Reply:
I am not sure I know what you mean, but this works:
```python
concert > args = [1, 2 ,3] * q.px                                                                                                             
concert > args                                                                                                                                
array([1, 2, 3]) <Unit('pixel')>
concert > args[0] = 10 * q.px                                                                                                                 
concert > args[0]                                                                                                                             
10 <Unit('pixel')>
concert > args                                                                                                                                
array([10,  2,  3]) <Unit('pixel')>
```

Reply:
But this not (independent of quantity etc.):

```python
from concert.base import Parameterizable

class MyParameterizable(Parameterizable):
   my_vector = Parameter()

   async def __ainit__(self):
      self._vec = [1, 2, 3]

   async def _get_my_vector(self):
      return self._my_vector

   async def _set_my_vector(self, vec):
      self._vec = vec
  
p = await MyParameterizable()
p.my_vector[1] = 5
```

Reply:
Ah that, well numpy managed to override []...

Reply:
I already thought to do this for the QuantityValue...

Reply:
Good, then I leave it up to you :wink: